### DISCUSSION TITLE
[Feature] SurrealDB driver

### DISCUSSION THREAD

**Discussion by ThatOneCalculator** [ponder-sh/ponder#878] [Idea / Feature Request]
A SurrealDB driver in addition to PostgreSQL and SQLite would be great!

**Comment by typedarray** [comment]
Thanks for opening. Could you share a bit more about why you want this? What problem would SurrealDB solve, and for what use cases would you prefer it over Postgres?

**Comment by ThatOneCalculator** [comment]
Definitely!

## Convenience in SurrealDB based projects

Probably the most obvious and immediate benefit would be that for projects already using SurrealDB (full disclosure, like the one I'm working on), having Ponder directly write to SurrealDB in a Ponder namespace would make things a lot simpler rather than having to query from the Ponder API -- as I'm sure you're aware, going Ponder --> SQL DB --> GQL API --> Backend API --> SurrealDB  is a lot less efficient than Ponder --> SurrealDB --> Backend API

## More data relation models

SurrealDB is a multi-model database, meaning that on top of schemafull and schemaless relational data, it also allows for document data and most importantly for Ponder, graph related data, meaning that Ponder could eventually have a way to lay out indexed data on a graph-style database within Surreal on top of a more traditional relational layout

## Live updates

SurrealDB makes it really easy to listen for live updates to the database without setting up complex systems like in Postgres

## Indexing & searching

SurrealDB provides a really great out-of-the-box solution for indexing a database for full-text semantic search, meaning that a real time search engine could be built incredibly easily off of a single Ponder instance connected to SurrealDB rather than having to injest and reindex that data onto a search engine like ElasticSearch or MeiliSearch.

## Inspection

SurrealDB has a very nice in-house tool called Surrealist, which makes viewing database models and data really simple and easy in both development and production. I've used visual tools for SQL-esque databases before and at least imo, they aren't as intuitive

## REST API
 
Even though Ponder already provides a GraphQL API, ingesting it into Surreal would also allow for developers to build out a custom REST API directly within SurrealDB without having to write their own backend if they prefer REST over GraphQL. 

## Running

In both development and production, SurrealDB is far simpler to configure and run compared to Postgres -- not *as* simple as SQLite, but far closer to the capabilities of Postgres (which is pretty hard to get running, especially locally on a dev machine).

**Comment by typedarray** [comment]
Thanks for the thorough response. We don't have any immediate plans to add more database drivers, but I'll move this to discussion and revisit down the road.

**Comment by ThatOneCalculator** [comment]
Gotcha. I'd have to talk to my team, but this is also something we could potentially sponsor/help with later on if you're open to that :)
